newton/geometry/inertia.py

```diff
@@ -485,3 +487,166 @@ def compute_shape_inertia(
+    corrected_inertia = inertia_array.copy()
+
+    # Check and correct mass
+    if mass < 0:
+        warnings.warn(f"Negative mass {mass} detected, setting to 0", stacklevel=2)
+        corrected_mass = 0.0
+        was_corrected = True
+    elif bound_mass is not None and mass < bound_mass and mass > 0:
+        warnings.warn(f"Mass {mass} is below bound {bound_mass}, clamping", stacklevel=2)
+        corrected_mass = bound_mass
+        was_corrected = True
+
+    # For zero mass, inertia should also be zero
+    if corrected_mass == 0.0:
+        if np.any(inertia_array != 0):
+            warnings.warn("Zero mass body should have zero inertia, correcting", stacklevel=2)
+            corrected_inertia = np.zeros((3, 3))
+            was_corrected = True
+        return corrected_mass, wp.mat33(corrected_inertia), was_corrected
+
+    # Check that inertia matrix is symmetric
+    if not np.allclose(inertia_array, inertia_array.T):
+        warnings.warn("Inertia matrix is not symmetric, making it symmetric", stacklevel=2)
+        corrected_inertia = (inertia_array + inertia_array.T) / 2
+        was_corrected = True
+
+    # Extract principal moments (diagonal elements)
+    Ixx, Iyy, Izz = corrected_inertia[0, 0], corrected_inertia[1, 1], corrected_inertia[2, 2]
+
+    # Check for negative diagonal elements
+    if Ixx < 0 or Iyy < 0 or Izz < 0:
+        warnings.warn(
+            f"Negative inertia diagonal elements detected: ({Ixx}, {Iyy}, {Izz}), taking absolute values", stacklevel=2
+        )
+        Ixx, Iyy, Izz = abs(Ixx), abs(Iyy), abs(Izz)
+        corrected_inertia[0, 0] = Ixx
+        corrected_inertia[1, 1] = Iyy
+        corrected_inertia[2, 2] = Izz
+        was_corrected = True
+
+    # Apply inertia bounds if specified
+    if bound_inertia is not None:
+        if Ixx < bound_inertia:
+            warnings.warn(f"Ixx {Ixx} is below bound {bound_inertia}, clamping", stacklevel=2)
+            Ixx = bound_inertia
+            corrected_inertia[0, 0] = Ixx
+            was_corrected = True
+        if Iyy < bound_inertia:
+            warnings.warn(f"Iyy {Iyy} is below bound {bound_inertia}, clamping", stacklevel=2)
+            Iyy = bound_inertia
+            corrected_inertia[1, 1] = Iyy
+            was_corrected = True
+        if Izz < bound_inertia:
+            warnings.warn(f"Izz {Izz} is below bound {bound_inertia}, clamping", stacklevel=2)
+            Izz = bound_inertia
+            corrected_inertia[2, 2] = Izz
+            was_corrected = True
+
+    # Check triangle inequality (principal moments must satisfy these constraints)
+    # For a physically valid inertia tensor: Ixx + Iyy >= Izz, Iyy + Izz >= Ixx, Izz + Ixx >= Iyy
+    violations = []
+    if Ixx + Iyy < Izz:
+        violations.append((0, 1, 2))  # Ixx + Iyy < Izz
+    if Iyy + Izz < Ixx:
+        violations.append((1, 2, 0))  # Iyy + Izz < Ixx
+    if Izz + Ixx < Iyy:
+        violations.append((2, 0, 1))  # Izz + Ixx < Iyy
```

Comment:
The `violations` list seems to be unused - can we just use a boolean to check if there are any violations?